CRM/Event/Form/Participant.php

```diff
@@ -327,8 +327,8 @@ public function preProcess() {
 
     $this->assign('participantMode', $this->_mode);
 
+    $this->assign('showFeeBlock', $this->_showFeeBlock);
     if ($this->_showFeeBlock) {
-      $this->assign('showFeeBlock', TRUE);
       $isMonetary = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_Event', $this->_showFeeBlock, 'is_monetary');
```

Comment:
This PR looks ok as-is just am a little confused at how this was coming up undefined in the tpl because there's a default set at line 273.

Reply:
I think it wasn't - it was showing as a pattern in php land - FWIW my thinking is that we can separate this craziness out by building on https://github.com/civicrm/civicrm-core/pull/20679 with a PR that creates a new FeeBlock form that is a subform of this one. There is so much to unravel & I've been trying to figure out how to break it down & I think that will work